types/math/math.go

```diff
@@ -0,0 +1,65 @@
+package math
+
+import (
+	"fmt"
+
+	"github.com/cockroachdb/apd/v2"
+
+	"github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// ParseNonNegativeDecimal parses a non-negative decimal or returns an error.
+func ParseNonNegativeDecimal(x string) (*apd.Decimal, error) {
+	res, _, err := apd.NewFromString(x)
+	if err != nil || res.Sign() < 0 {
+		return nil, errors.Wrap(errors.ErrInvalidRequest, fmt.Sprintf("expected a non-negative decimal, got %s", x))
+	}
+
+	return res, nil
+}
+
+// ParsePositiveDecimal parses a positive decimal or returns an error.
+func ParsePositiveDecimal(x string) (*apd.Decimal, error) {
+	res, _, err := apd.NewFromString(x)
+	if err != nil || res.Sign() <= 0 {
+		return nil, errors.Wrap(errors.ErrInvalidRequest, fmt.Sprintf("expected a positive decimal, got %s", x))
+	}
+
+	return res, nil
+}
+
+// DecimalString prints x as a floating point string.
+func DecimalString(x *apd.Decimal) string {
+	return x.Text('f')
+}
+
+var exactContext = apd.Context{
+	Precision:   0,
+	MaxExponent: apd.MaxExponent,
+	MinExponent: apd.MinExponent,
+	Traps:       apd.DefaultTraps | apd.Inexact | apd.Rounded,
+}
+
+// Add adds x and y and stores the result in res with arbitrary precision or returns an error.
+func Add(res, x, y *apd.Decimal) error {
```

Comment:
I don't think we're gonna need this function, ultimately we should be using `Add` and `Sub` methods from the `Dec struct` in the group module too, as described in the issue description.